tests/basic/sharedv4_test.go

```diff
@@ -130,9 +130,14 @@ var _ = Describe("{Sharedv4Functional}", func() {
 				})
 
 				Step(fmt.Sprintf("validate counter are active for %s", ctx.App.Key), func() {
-					counters := getAppCounters(apiVol, attachedNode, counterCollectionInterval)
-					activePods := getActivePods(counters)
-					Expect(len(activePods)).To(Equal(numPods))
+					var counters map[string]appCounter
+					Eventually(func() bool {
+						counters = getAppCounters(apiVol, attachedNode, counterCollectionInterval)
+						activePods := getActivePods(counters)
+						return (len(activePods) == numPods)
+					}, 3*time.Minute, 10*time.Second).Should(BeTrue(),
```

Comment:
is 3 min enough? Please add some comments about why we need to keep retrying here.